src/core/shape_inference/include/string_tensor_unpack_shape_inference.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+
+#include "openvino/op/string_tensor_unpack.hpp"
+#include "shape_infer_type_utils.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+namespace util {
+static Tensor get_string_tensor(const Node* op, size_t idx, const ITensorAccessor& tensor_accessor) {
+    if (auto t = tensor_accessor(idx)) {
+        return t;
+    } else {
+        const auto& constant = as_type_ptr<opset1::Constant>(op->get_input_node_shared_ptr(idx));
+        NODE_VALIDATION_CHECK(op, constant != nullptr, "Static shape inference lacks constant data on port ", idx);
+        return constant->get_tensor_view();
+    }
+}
+}  // namespace util
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const StringTensorUnpack* op,
+                                 const std::vector<TShape>& input_shapes,
+                                 const ITensorAccessor& tensor_accessor = make_tensor_accessor()) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1);
+    const auto& data_shape = input_shapes[0];
+    auto output_shapes = std::vector<TRShape>(3);
+
+    // output 1 and 2: begins and ends
+    output_shapes[0] = data_shape;
+    output_shapes[1] = data_shape;
+
+    // output 3: symbols
+    const auto string_data = util::get_string_tensor(op, 0, tensor_accessor);
+    if (string_data) {
+        uint64_t string_count = 1;
+        for (uint64_t i = 0; i < data_shape.size(); i++) {
+            string_count *= data_shape[i].get_length();
+        }
+        const auto tensor_data = string_data.data<std::string>();
+        uint64_t total_length = 0;
+        for (size_t i = 0; i < string_count; ++i)
+            total_length += (*(tensor_data + i)).length();
```

Comment:
```suggestion
auto total_length = std::accumulate(tensor_data, tensor_data + string_data->get_size(), 0, [](const std::string& s){
    return s.length();
});
```

Check if `tensor::get_byte_size()` value can be used instead?

Reply:
I tried, but couldn't find an exact pattern of how byte size increases with number of characters. The result would be imprecise at best.

Reply:
True, byte size will return string object size in bytes, not number of characters.